src/main/scala/sparkz/core/network/PeerConnectionHandler.scala

```diff
@@ -204,7 +204,7 @@ class PeerConnectionHandler(val settings: NetworkSettings,
               case MaliciousBehaviorException(msg) =>
                 log.warn(s"Banning peer for malicious behaviour($msg): ${connectionId.toString}")
                 //peer will be added to the blacklist and the network controller will send CloseConnection
```

Comment:
this comment seems outdated now

Reply:
seems ok after the last changes with disconnectPenalty.
@paologalligit pls check it

Reply:
actually, Ivan was quite right: although the intention was to ban the peer with the DisconnectPenalty temporarily, the implementation was bugged. To penalize the peer we need the penalty score to go over the threshold, while the DisconnectPenalty is reaching the exact score of the threshold resulting in a penalization but not a ban. Pushing the fix for it; thanks